src/server.py

```diff
@@ -78,6 +81,37 @@ def get_date_range(dates):
                 'endDate': max(dates).strftime('%Y-%m-%d %H:%M')}
 
 
+def valve(head_rev, pushlog_id):
+    """Determine which kind of job should been returned"""
+    low_job_return = True
+    COUNTER.increase_the_counter()
+    request_list = []
+    try:
+        request_list = session.query(TaskRequests.head_rev, TaskRequests.pushlog_id,
+                                     TaskRequests.low_job_return).all()
+    except Exception:
+        session.rollback()
```

Comment:
we rollback here is because it will block other operations of this session(this session will become unusable) when an exception been raised and haven't been rollback after that. We do the same thing in updatedb.py before, but I forget to mark that related document when I read it. Please give me some time to fine that document if you need it :)

hmmm, where should we use that decrease_the_counter()? If the session rollback, we will just add that push into the database. Should we checkout the whole table again for removing those duplicate items?